From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: deepak1556 <hop2deep@gmail.com>
Date: Tue, 2 Feb 2021 22:42:32 -0800
Subject: feat: export gin's PageAllocator

In order for memory allocation in the node processes to be
correctly tagged with MAP_JIT we need to use gin's page allocator instead
of the default v8 allocator.

diff --git a/gin/public/v8_platform.h b/gin/public/v8_platform.h
index da2aeb2f2da84fe47d5cc7d721f8d3dade0c0972..4ddad9065782fad5927bc0481ad772014af0d689 100644
--- a/gin/public/v8_platform.h
+++ b/gin/public/v8_platform.h
@@ -23,6 +23,7 @@ class GIN_EXPORT V8Platform : public v8::Platform {
 // Some configurations do not use page_allocator.
 #if BUILDFLAG(USE_PARTITION_ALLOC)
   v8::PageAllocator* GetPageAllocator() override;
+  static v8::PageAllocator* PageAllocator();
   void OnCriticalMemoryPressure() override;
 #endif
   std::shared_ptr<v8::TaskRunner> GetForegroundTaskRunner(
diff --git a/gin/v8_platform.cc b/gin/v8_platform.cc
index 319dca5afc2f5d19a0c8c57e0d7192531050be04..a33d1451d542164ddef1df1c624810827ac1fa99 100644
--- a/gin/v8_platform.cc
+++ b/gin/v8_platform.cc
@@ -431,6 +431,10 @@ v8::PageAllocator* V8Platform::GetPageAllocator() {
   return g_page_allocator.Pointer();
 }
 
+v8::PageAllocator* V8Platform::PageAllocator() {
+  return g_page_allocator.Pointer();
+}
+
 void V8Platform::OnCriticalMemoryPressure() {
 // We only have a reservation on 32-bit Windows systems.
 // TODO(bbudge) Make the #if's in BlinkInitializer match.
